src/HttpBodyStream.php

```diff
@@ -0,0 +1,169 @@
+<?php
+namespace Legionth\React\Http;
+
+use Psr\Http\Message\StreamInterface;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\WritableStreamInterface;
+use Evenement\EventEmitter;
+use React\Stream\Util;
+
+/**
+ * Uses a StreamInterface from PSR-7 and a ReadableStreamInterface from ReactPHP
+ * to use PSR-7 objects and use ReactPHP streams
+ */
+class HttpBodyStream extends EventEmitter implements StreamInterface, ReadableStreamInterface
+{
+    private $input;
+    private $closed = false;
+    private $encoder;
+
+    public function __construct(ReadableStreamInterface $input, ReadableStreamInterface $encoder = null)
+    {
+        $this->input = $input;
+
+        if ($encoder === null) {
+            $encoder = new ChunkedEncoderStream($this->input);
+        }
```

Comment:
This makes sense from a SOLID standpoint (DI), but it looks kind of odd from a consumer perspective. Perhaps the constructor could explain the *why*?